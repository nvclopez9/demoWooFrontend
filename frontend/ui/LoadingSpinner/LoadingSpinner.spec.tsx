import { render, screen } from '@woographql/testing';
import { LoadingSpinner } from '.';

describe('LoadingSpinner component', () => {
  it('renders correctly with default props', () => {
    render(<LoadingSpinner />);

    // Check that the spinner is rendered
    const spinner = screen.getByRole('status', { name: 'Loading...' });
    expect(spinner).toBeInTheDocument();

    // Check that the loading text is rendered
    const loadingText = screen.getByText('Loading...');
    expect(loadingText).toBeInTheDocument();
  });

  it('renders correctly with noText prop', () => {
    render(<LoadingSpinner noText />);

    // Check that the spinner is rendered
    const spinner = screen.getByRole('status', { name: 'Loading...' });
    expect(spinner).toBeInTheDocument();

    // Check that the loading text is not rendered
    const loadingText = screen.queryByText('Loading...');
    expect(loadingText).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<LoadingSpinner />);
    expect(baseElement).toMatchSnapshot();
  });
});
