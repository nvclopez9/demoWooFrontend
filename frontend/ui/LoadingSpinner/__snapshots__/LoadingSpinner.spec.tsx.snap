// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component should match snapshot 1`] = `
<body>
  <div>
    <div
      aria-label="Loading..."
      class="relative flex items-center justify-center space-x-2 "
      role="status"
    >
      <svg
        class="h-6 w-6 animate-spin stroke-amber-600"
        viewBox="0 0 256 256"
      >
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="128"
          x2="128"
          y1="32"
          y2="64"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="195.9"
          x2="173.3"
          y1="60.1"
          y2="82.7"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="224"
          x2="192"
          y1="128"
          y2="128"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="195.9"
          x2="173.3"
          y1="195.9"
          y2="173.3"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="128"
          x2="128"
          y1="224"
          y2="192"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="60.1"
          x2="82.7"
          y1="195.9"
          y2="173.3"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="32"
          x2="64"
          y1="128"
          y2="128"
        />
        <line
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="60.1"
          x2="82.7"
          y1="60.1"
          y2="82.7"
        />
      </svg>
      <span
        class="text-xs font-medium text-gray-500"
      >
        Loading...
      </span>
    </div>
  </div>
</body>
`;
